src/openstudio_lib/GeometryPreviewView.cpp

```diff
@@ -156,7 +180,11 @@ void PreviewWebView::onLoadFinished(bool ok) {
     std::function<void(double)> updatePercentage = std::bind(&PreviewWebView::onTranslateProgress, this, _1);
     //ThreeScene scene = modelToThreeJS(m_model.clone(true).cast<model::Model>(), true, updatePercentage); // triangulated
 
+    // qDebug() << "ThreeJSForwardTranslator";
+
     model::ThreeJSForwardTranslator ft;
+    m_includeGeometryDiagnostics = m_geometryDiagnosticsBox->isChecked();
+    ft.setIncludeGeometryDiagnostics(m_includeGeometryDiagnostics);
```

Comment:
See original comments https://github.com/openstudiocoalition/OpenStudioApplication/pull/629#discussion_r1308156144

@macumber said:



> Is there a downside to just always including the geometry diagnostics on this page? Is it significantly slower?

Reply:
After I implemented caching, it's not that bad, but yes, it's slightly slower. See benchmarks here: https://github.com/NREL/OpenStudio/pull/4843#discussion_r1156098275

With caching:

| Test case                         | Release | Release   | Debug   | Debug     |
|-----------------------------------|---------|-----------|---------|-----------|
|                                   | No Diag | With Diag | No Diag | With Diag |
| RefBldgOutPatientNew2004_Chicago  | 1.39    | 1.4       | 10.09   | 10.19     |
| 7_7_Windows_Complete              | 3.79    | 3.95      | 27.17   | 28.33     |